test/unit/modules/blocks/utils.js

```diff
@@ -13,268 +13,582 @@
+		);
 
-			it('should assign db');
+		modulesStub = {
+			blocks: {
+				lastBlock: {
+					get: sinonSandbox
+						.stub()
+						.returns({ id: '9314232245035524467', height: 1 }),
+					set: sinonSandbox
+						.stub()
+						.returns({ id: '9314232245035524467', height: 1 }),
+				},
+				utils: {
+					readDbRows: blocksUtilsModule.readDbRows,
+				},
+			},
+		};
+
+		blocksUtilsModule.onBind(modulesStub);
+
+		__private = BlocksUtils.__get__('__private');
+		__private.loaded = false;
+
+		library = BlocksUtils.__get__('library');
+		modules = BlocksUtils.__get__('modules');
+	});
 
-			it('should assign dbSequence');
+	afterEach(() => {
+		sinonSandbox.reset();
+	});
 
-			it('should assign genesisblock');
+	describe('constructor', () => {
+		describe('library', () => {
+			it('should assign logger', () => {
+				expect(library.logger).to.eql(loggerStub);
+			});
 
 			describe('should assign logic', () => {
-				it('should assign block');
+				it('should assign account', () => {
+					expect(library.logic.account).to.eql(accountMock);
+				});
+
+				it('should assign block', () => {
+					expect(library.logic.block).to.eql(blockMock);
+				});
 
-				it('should assign transaction');
+				it('should assign transaction', () => {
+					expect(library.logic.transaction).to.eql(transactionMock);
+				});
 			});
-		});
 
-		it('should set self to this');
+			it('should assign dbSequence', () => {
+				expect(library.dbSequence).to.eql(modulesLoader.scope.dbSequence);
+			});
 
-		it(
-			'should call library.logger.trace with "Blocks->Utils: Submodule initialized."'
-		);
+			it('should assign db', () => {
+				expect(library.db).to.eql(dbStub);
+			});
+		});
 
-		it('should return self');
+		it('should call library.logger.trace with "Blocks->Utils: Submodule initialized."', () => {
+			expect(loggerStub.trace.args[0][0]).to.equal(
+				'Blocks->Utils: Submodule initialized.'
+			);
+		});
+
+		it('should return self', () => {
+			expect(blocksUtilsModule).to.be.an('object');
+			expect(blocksUtilsModule.readDbRows).to.be.a('function');
+		});
 	});
 
 	describe('readDbRows', () => {
-		it('should loop through all given rows');
-
-		it('should call library.logic.block.dbRead with every row');
-
-		describe('when block with id does not exist', () => {
-			it('should not return the block');
+		it('should transform a full_blocks_list view row into a block object', done => {
+			var blockObject = blocksUtilsModule.readDbRows(viewRow_full_blocks_list);
+
+			expect(blockObject).to.be.an('array');
+			expect(blockObject[0]).to.be.an('object');
+			expect(blockObject[0].id).to.equal('13068833527549895884');
+			expect(blockObject[0].height).to.equal(3);
+			expect(blockObject[0].transactions).to.be.an('array');
+			expect(blockObject[0].transactions[0]).to.be.an('object');
+			expect(blockObject[0].transactions[0].id).to.equal('6950874693022090568');
+			expect(blockObject[0].transactions[0].type).to.equal(0);
+			done();
```

Comment:
We are not doing any async functionality in this test so why to use `done` pattern.

Reply:
Done.